### ISSUE TITLE
Cant send email on the server itself

### ISSUE THREAD

**Issue by geilmaker** [docker-mailserver/docker-mailserver#2557]
# Subject
I cant send emails on the server from one mailbox to another.

## Description
Hello everybody :)
I am trying to get this mailserver running but I am failing on some point and I dont really know where... I just set the server up with the documentation and I tried sending E-Mails to GMail, that didnt work for authentication reasons or sth like that... Now I created a second mail account on the server and try to send e-mails between both of them, but they dont appear in the others mailbox. I dont know why, does anybody know what might happen there? Because its all on one server it cant be a network problem like ports/dns records, aight?

``` BASH
mailserver    | Apr 19 12:03:51 mail amavis[1448]: (01448-01) Passed CLEAN {RelayedInbound}, [91.0.109.71]:52295 [91.0.109.71] <me@mail.domain.de> -> <test@mail.domain.de>, Queue-ID: 3E2B513BF21, Message-ID: <0226d13f-9cf3-a531-ac7e-2d2b732d713c@mail.domain.de>, mail_id: lhqkVefUn2F2, Hits: 2.175, size: 1236, queued_as: 1A60113BF2D, 1726 ms
mailserver    | Apr 19 12:03:51 mail postfix/smtp[1517]: 3E2B513BF21: to=<test@mail.domain.de>, relay=127.0.0.1[127.0.0.1]:10024, delay=1.9, delays=0.14/0.02/0.01/1.7, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 1A60113BF2D)
mailserver    | Apr 19 12:03:51 mail postfix/qmgr[1431]: 3E2B513BF21: removed
mailserver    | Apr 19 12:03:51 mail postfix/local[1527]: 1A60113BF2D: to=<test@mail.domain.de>, relay=local, delay=0.03, delays=0.01/0.01/0/0.01, dsn=5.1.1, status=bounced (unknown user: "test")
```
Thats the log after sending the email, I hope its all the important part, if not please just ask.

When I go for ./setup.sh email list I get those 2 mailboxes I created
``` BASH
* me@mail.domain.de
* test@mail.domain.de
```

The sent emails appear in the sent folder of the mailbox but the other mailbox does not receive them.

Thanks for any help, have a good one!

**Comment by casperklein** [comment]
> because its all on one server it cant be a network problem like ports/dns records, aight?

correct.

You can see in the logs, that the mail was bounced because the user could not be found:

`mailserver    | Apr 19 12:03:51 mail postfix/local[1527]: 1A60113BF2D: to=<test@mail.domain.de>, relay=local, delay=0.03, delays=0.01/0.01/0/0.01, dsn=5.1.1, status=bounced (unknown user: "test")`.

What does your `docker-compose.yml` file look like? Also you can enable the [debug mode](https://docker-mailserver.github.io/docker-mailserver/v10.5/config/environment/#dms_debug) and post the whole output after DMS has started.

**Comment by geilmaker** [comment]
Hey!
Thanks for the reply!

I used the following command:
```BATCH
docker-compose --log-level DEBUG up
```
and got the output:
```BATCH
Attaching to mailserver
mailserver    | [ TASKLOG ]  Welcome to docker-mailserver 10.5.0
mailserver    | [ TASKLOG ]  Initializing setup
mailserver    | [ TASKLOG ]  Checking configuration
mailserver    | [ TASKLOG ]  Configuring mail server
mailserver    | [ TASKLOG ]  Post-configuration checks
mailserver    | [ TASKLOG ]  Starting daemons & mail server
mailserver    | cron: started
mailserver    | rsyslog: started
mailserver    | dovecot: started
mailserver    | update-check: started
mailserver    | opendkim: started
mailserver    | opendmarc: started
mailserver    | postgrey: started
mailserver    | postfix: started
mailserver    | clamav: started
mailserver    | changedetector: started
mailserver    | amavis: started
mailserver    | [ TASKLOG ]  mail.domain.de is up and running
mailserver    | Apr 19 22:09:41 mail amavis[524]: starting. /usr/sbin/amavisd-new at mail.domain.de amavisd-new-2.11.1 (20181009), Unicode aware, LC_CTYPE="C.UTF-8"
mailserver    | Apr 19 22:09:41 mail amavis[524]: perl=5.032001, user=, EUID: 112 (112);  group=, EGID: 114 114 (114 114)
mailserver    | Apr 19 22:09:41 mail amavis[524]: Net::Server: Group Not Defined.  Defaulting to EGID '114 114'
mailserver    | Apr 19 22:09:41 mail amavis[524]: Net::Server: User Not Defined.  Defaulting to EUID '112'
mailserver    | Apr 19 22:09:41 mail amavis[524]: No ext program for   .zoo, tried: zoo
mailserver    | Apr 19 22:09:41 mail amavis[524]: No ext program for   .doc, tried: ripole
mailserver    | Apr 19 22:09:41 mail amavis[524]: No decoder for       .F
mailserver    | Apr 19 22:09:41 mail amavis[524]: No decoder for       .doc
mailserver    | Apr 19 22:09:41 mail amavis[524]: No decoder for       .zoo
mailserver    | Apr 19 22:09:41 mail amavis[524]: Using primary internal av scanner code for ClamAV-clamd
mailserver    | Apr 19 22:09:41 mail amavis[524]: Found secondary av scanner ClamAV-clamscan at /usr/bin/clamscan
mailserver    | Apr 19 22:09:42 mail postfix/master[1428]: daemon started -- version 3.5.6, configuration /etc/postfix
```

My docker-compose.yml file:
```BATCH
version: '3.8'

services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver
    hostname: mail
    domainname: domain.de
    ports:
      - "25:25"
      - "143:143"
      - "587:587"
      - "993:993"
    volumes:
      - ./docker-data/dms/mail-data/:/var/mail/
      - ./docker-data/dms/mail-state/:/var/mail-state/
      - ./docker-data/dms/mail-logs/:/var/log/mail/
      - ./docker-data/dms/config/:/tmp/docker-mailserver/
      - /home/nginx/certbot/conf:/etc/letsencrypt
      - /etc/localtime:/etc/localtime:ro
    environment:
      - ENABLE_AMAVIS=1
      - ENABLE_CLAMAV=1
      #- ENABLE_FAIL2BAN=1
      - SPOOF_PROTECTION=1
      - POSTMASTER_ADDRESS=myprivatemail@gmail.com
      - POSTFIX_INET_PROTOCOLS=ipv4
      - DOVECOT_INET_PROTOCOLS=ipv4
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_POSTGREY=1
      - ENABLE_SASLAUTHD=0
      - SSL_TYPE=letsencrypt
      - ONE_DIR=1
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
    restart: unless-stopped
```

**Comment by geilmaker** [comment]
And the following is the complete log when sending an email:

```BATCH
mailserver    | Apr 19 22:15:19 mail postfix/submission/smtpd[2246]: connect from <numbers>.dip0.t-ipconnect.de[91.x.x.x]
mailserver    | Apr 19 22:15:19 mail postfix/submission/smtpd[2246]: Anonymous TLS connection established from <numbers>.dip0.t-ipconnect.de[91.x.x.x]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature RSA-PSS (2048 bits) server-digest SHA256
mailserver    | Apr 19 22:15:19 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
mailserver    | Apr 19 22:15:19 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
mailserver    | Apr 19 22:15:19 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
mailserver    | Apr 19 22:15:19 mail postfix/submission/smtpd[2246]: 6D4BF13BF2E: client=<numbers>.dip0.t-ipconnect.de[91.x.x.x], sasl_method=PLAIN, sasl_username=me@mail.domain.de
mailserver    | Apr 19 22:15:19 mail postfix/sender-cleanup/cleanup[2256]: 6D4BF13BF2E: message-id=<926cfaa7-bbd3-d706-94cb-b6eed10cbfc9@mail.domain.de>
mailserver    | Apr 19 22:15:19 mail postfix/sender-cleanup/cleanup[2256]: 6D4BF13BF2E: replace: header MIME-Version: 1.0 from <numbers>.dip0.t-ipconnect.de[91.x.x.x]; from=<me@mail.domain.de> to=<test@mail.domain.de> proto=ESMTP helo=<[192.168.178.133]>: Mime-Version: 1.0
mailserver    | Apr 19 22:15:19 mail opendkim[229]: 6D4BF13BF2E: DKIM-Signature field added (s=mail, d=mail.domain.de)
mailserver    | Apr 19 22:15:19 mail postfix/qmgr[1431]: 6D4BF13BF2E: from=<me@mail.domain.de>, size=371, nrcpt=1 (queue active)
mailserver    | Apr 19 22:15:19 mail postfix/submission/smtpd[2246]: disconnect from <numbers>.dip0.t-ipconnect.de[91.x.x.x] ehlo=2 starttls=1 auth=1 mail=1 rcpt=1 data=1 quit=1 commands=8
mailserver    | Apr 19 22:15:19 mail dovecot: imap-login: Login: user=<me@mail.domain.de>, method=PLAIN, rip=91.x.x.x, lip=172.x.x.x, mpid=2260, TLS, session=<h3IwiQfdBPFbAG1H>
mailserver    | Apr 19 22:15:20 mail dovecot: imap-login: Login: user=<me@mail.domain.de>, method=PLAIN, rip=91.x.x.x, lip=172.x.x.x, mpid=2263, TLS, session=<9943iQfdBfFbAG1H>
mailserver    | Apr 19 22:15:21 mail postfix/smtpd[2265]: connect from localhost[127.0.0.1]
mailserver    | Apr 19 22:15:21 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
mailserver    | Apr 19 22:15:21 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
mailserver    | Apr 19 22:15:21 mail postfix/smtpd[2265]: 161AD13BF2F: client=localhost[127.0.0.1]
mailserver    | Apr 19 22:15:21 mail postfix/cleanup[2266]: 161AD13BF2F: message-id=<926cfaa7-bbd3-d706-94cb-b6eed10cbfc9@mail.domain.de>
mailserver    | Apr 19 22:15:21 mail postfix/qmgr[1431]: 161AD13BF2F: from=<me@mail.domain.de>, size=1823, nrcpt=1 (queue active)
mailserver    | Apr 19 22:15:21 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
mailserver    | Apr 19 22:15:21 mail amavis[1449]: (01449-01) Passed CLEAN {RelayedInbound}, [91.x.x.x]:61699 [91.x.x.x] <me@mail.domain.de> -> <test@mail.domain.de>, Queue-ID: 6D4BF13BF2E, Message-ID: <926cfaa7-bbd3-d706-94cb-b6eed10cbfc9@mail.domain.de>, mail_id: Tq4S6DD69O_n, Hits: 2.175, size: 1243, queued_as: 161AD13BF2F, 1482 ms
mailserver    | Apr 19 22:15:21 mail postfix/smtp[2257]: 6D4BF13BF2E: to=<test@mail.domain.de>, relay=127.0.0.1[127.0.0.1]:10024, delay=1.7, delays=0.16/0.03/0.03/1.5, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 161AD13BF2F)
mailserver    | Apr 19 22:15:21 mail postfix/qmgr[1431]: 6D4BF13BF2E: removed
mailserver    | Apr 19 22:15:21 mail postfix/local[2267]: 161AD13BF2F: to=<test@mail.domain.de>, relay=local, delay=0.02, delays=0.01/0.01/0/0.01, dsn=5.1.1, status=bounced (unknown user: "test")
mailserver    | Apr 19 22:15:21 mail postfix/cleanup[2266]: 1A7F213BF2E: message-id=<20220419201521.1A7F213BF2E@mail.domain.de>
mailserver    | Apr 19 22:15:21 mail postfix/bounce[2268]: 161AD13BF2F: sender non-delivery notification: 1A7F213BF2E
mailserver    | Apr 19 22:15:21 mail postfix/qmgr[1431]: 1A7F213BF2E: from=<>, size=3797, nrcpt=1 (queue active)
mailserver    | Apr 19 22:15:21 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
mailserver    | Apr 19 22:15:21 mail postfix/qmgr[1431]: 161AD13BF2F: removed
mailserver    | Apr 19 22:15:21 mail postfix/local[2267]: 1A7F213BF2E: to=<me@mail.domain.de>, relay=local, delay=0, delays=0/0/0/0, dsn=5.1.1, status=bounced (unknown user: "me")
mailserver    | Apr 19 22:15:21 mail postfix/qmgr[1431]: 1A7F213BF2E: removed
```

When I use ./setup.sh email list I am getting both mailboxes, so I dont know why I get
```BATCH
unknown user: "test"
```
and
```BATCH
unknown user: "me"
```

**Comment by casperklein** [comment]
Try setting this env in your docker-compose.yml:

 `OVERRIDE_HOSTNAME=mail.domain.de`

Don't forget to properly restart DMS with `docker-compose down/up`

**Comment by geilmaker** [comment]
Thanks for the advise, I added the line to the environment variables inside the docker-compose.yml, but I still got the same errors (bounced, unknown user) and same behavior (not receiving the email)

Yes, I restart the container every time with docker-compose down and docker-compose up.

**Comment by casperklein** [comment]
> mailserver    | Apr 19 22:15:19 mail postfix/trivial-rewrite[2255]: warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains

This is not normal and likely your problem. That's why I suggested to test with `OVERRIDE_HOSTNAME`.
Do you have setup aliases or something?

**Comment by polarathene** [comment]
> I used the following command:
> 
> ```batchfile
> docker-compose --log-level DEBUG up
> ```

You did the following mentioned in [`docker-compose` CLI docs](https://docs.docker.com/compose/reference/), but note that is deprecated for one:

> _--log-level LEVEL           DEPRECATED and not working from 2.0 - Set log level (DEBUG, INFO, WARNING, ERROR, CRITICAL)_

And also that's not what the [docs link provided to you](https://docker-mailserver.github.io/docker-mailserver/v10.5/config/environment/#dms_debug) instructed. You were meant to use `DMS_DEBUG=1` (_note that this will change to [`LOG_LEVEL=debug` from v11](https://docker-mailserver.github.io/docker-mailserver/edge/config/environment/#log_level) onwards_).

---

One suggestion instead of `OVERRIDE_HOSTNAME` is to use `hostname: mail.domain.de` without the `domainname` property in your `docker-compose.yml`.

That aside, you want to figure out why your configuration is causing `warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains` to be logged. Perhaps start with a minimal configuration on a new container instance (new host volume data, start from scratch), and get that working first, then add your other config modifications to see if something else is causing the unexpected config.

**Comment by geilmaker** [comment]
Okay, thank you!
Now I have sth to work on, I'll text you again if I'm really stuck or get it to work.

I have no aliases or sth, I only added those 2 mailboxes and the DKIM keys with the setup.sh.

**Comment by geilmaker** [comment]
Hello :)
I found a lot of help inside the internet for that postfix problem, I could fix that by editing the main.cf file inside the docker container at /etc/postfix. Inside the file, I removed $myhostname at the key mydestination...
After that I was even able to send EMails to Gmail, but after a restart of docker-compose I would need to edit that again... So there should be a way to configure that inside docker compose, aight? :)

**Comment by geilmaker** [comment]
And if anybody knows, I would appreciate if someone could explain what I changed there and why it is working with that change :)

**Comment by polarathene** [comment]
> Inside the file, I removed $myhostname at the key mydestination...

This is a common workaround users have been choosing to do, but unfortunately it doesn't help us identify what is causing this to be required in the first place.

> but after a restart of docker-compose I would need to edit that again... So there should be a way to configure that inside docker compose

Another currently open issue that uses this workaround [suggests you maintain it via `user-patches.sh`](https://github.com/docker-mailserver/docker-mailserver/issues/2545#issuecomment-1100639118). 

We won't be doing this by default or as a configurable as we'd rather understand why this is happening for some users. It seems the first step towards that is probably to make it easier to reproduce, which is something I'd like to contribute for troubleshooting in general so we can have a common setup to compare to a users problematic one and work from there.

> And if anybody knows, I would appreciate if someone could explain what I changed there and why it is working with that change :)

This was mentioned already for you earlier in the discussion. Postfix was logging this: `warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains`, which is a warning that you should not have logged if configured correctly.

I will try to reproduce your setup by the weekend, it may be something wrong on our side, or specific to your configuration values or environment.

**Comment by geilmaker** [comment]
Okay, thank you, Ill try this script later on!

I tried the following configuration and used docker-compose up as root user, but I get the same error:
```BATCH
version: '3.8'

services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: mailserver
    hostname: mail.domain.de
    ports:
      - "25:25"
      - "143:143"
      - "587:587"
      - "993:993"
    volumes:
      - ./docker-data/dms/mail-data/:/var/mail/
      - ./docker-data/dms/mail-state/:/var/mail-state/
      - ./docker-data/dms/mail-logs/:/var/log/mail/
      - ./docker-data/dms/config/:/tmp/docker-mailserver/
      - /home/nginx/certbot/conf:/etc/letsencrypt
      - /etc/localtime:/etc/localtime:ro
    environment:
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_CLAMAV=1
      #- ENABLE_FAIL2BAN=1
      - ENABLE_POSTGREY=1
      - ENABLE_SASLAUTHD=0
      - ONE_DIR=1
      - SSL_TYPE=letsencrypt
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
    restart: always
```

That is the default configuration from https://github.com/docker-mailserver/docker-mailserver with SSL TYPE added, the volume for the certs added and commented FAIL2BAN out.
So on my system it doesnt even work with the basic setup + SSL as root user, hope it might help you!

Please let me know if I could help you figure out the problem on my "buggy" system!

My Docker Version:
```BATCH
Docker version 20.10.14, build a224086
```

**Comment by geilmaker** [comment]
Hello everybody!
I figured sth out.... I had the mailserver running on mail.domain.de and so I created my mailboxes like "me@mail.domain.de", but at that moment this domain (mail.domain.de) gets added as vhost and from that on I get this warning, like mentioned in the warning text. So I removed my data folder and started all from scratch, this time I added "me@domain.de", so this domain got added as vhost and it works just fine! There seems to be a problem when adding mailboxes with the same domain/host the mailserver is running on, is that intentionate?
Atm I can just receive mails from GMail and cannot send any to gmail, because of authentication reasons, even tho I setup spf and DKIM, but maybe it just need some time...

**Comment by casperklein** [comment]
I cannot confirm your finding:

```bash
DMS_GITHUB_URL='https://raw.githubusercontent.com/docker-mailserver/docker-mailserver/master'
wget "${DMS_GITHUB_URL}/docker-compose.yml"
wget "${DMS_GITHUB_URL}/mailserver.env"
wget "${DMS_GITHUB_URL}/setup.sh"
chmod a+x ./setup.sh
./setup.sh email add foo@mail.example.com foo
docker-compose up
```

```
[+] Running 2/0
 ⠿ Network build_default  Created                                                                                                                                                                                                       0.0s
 ⠿ Container mailserver   Created                                                                                                                                                                                                       0.0s
Attaching to mailserver
mailserver  | [ TASKLOG ]  Welcome to docker-mailserver 10.5.0
mailserver  | [ TASKLOG ]  Initializing setup
mailserver  | [ TASKLOG ]  Checking configuration
mailserver  | [ TASKLOG ]  Configuring mail server
mailserver  | [ WARNING ]  No DKIM key provided. Check the documentation on how to get your keys.
mailserver  | [ WARNING ]  (INSECURE!) SSL configured with plain text access. DO NOT USE FOR PRODUCTION DEPLOYMENT.
mailserver  | [ WARNING ]  Spamassassin is disabled. You can enable it with 'ENABLE_SPAMASSASSIN=1'
mailserver  | [ WARNING ]  Clamav is disabled. You can enable it with 'ENABLE_CLAMAV=1'
mailserver  | [ TASKLOG ]  Post-configuration checks
mailserver  | [ TASKLOG ]  Starting daemons & mail server
mailserver  | cron: started
mailserver  | rsyslog: started
mailserver  | dovecot: started
mailserver  | update-check: started
mailserver  | opendkim: started
mailserver  | opendmarc: started
mailserver  | postfix: started
mailserver  | changedetector: started
mailserver  | amavis: started
mailserver  | [ TASKLOG ]  mail.example.com is up and running
mailserver  | Apr 21 13:49:55 mail amavis[230]: starting. /usr/sbin/amavisd-new at mail.example.com amavisd-new-2.11.1 (20181009), Unicode aware, LC_CTYPE="C.UTF-8"
mailserver  | Apr 21 13:49:55 mail amavis[230]: perl=5.032001, user=, EUID: 112 (112);  group=, EGID: 114 114 (114 114)
mailserver  | Apr 21 13:49:55 mail amavis[230]: Net::Server: Group Not Defined.  Defaulting to EGID '114 114'
mailserver  | Apr 21 13:49:55 mail amavis[230]: Net::Server: User Not Defined.  Defaulting to EUID '112'
mailserver  | Apr 21 13:49:55 mail amavis[230]: No ext program for   .zoo, tried: zoo
mailserver  | Apr 21 13:49:55 mail amavis[230]: No ext program for   .doc, tried: ripole
mailserver  | Apr 21 13:49:55 mail amavis[230]: No decoder for       .F
mailserver  | Apr 21 13:49:55 mail amavis[230]: No decoder for       .doc
mailserver  | Apr 21 13:49:55 mail amavis[230]: No decoder for       .zoo
mailserver  | Apr 21 13:49:58 mail postfix/master[1406]: daemon started -- version 3.5.6, configuration /etc/postfix
```

DMS starts just fine, no warning about listing the domain in both, `mynetworks` and `virtual_mailbox_domains`.

**Comment by polarathene** [comment]
> DMS starts just fine, no warning about listing the domain in both, `mynetworks` and `virtual_mailbox_domains`.

Might be a different outcome with `hostname: mail.example.com` perhaps? (_which I think @geilmaker has been using since advised to_)

https://github.com/docker-mailserver/docker-mailserver/blob/377bdb390e87868617d45fe2c086a93a168be0fa/target/scripts/helpers/postfix.sh#L8

https://github.com/docker-mailserver/docker-mailserver/blob/377bdb390e87868617d45fe2c086a93a168be0fa/target/scripts/helpers/postfix.sh#L19-L34

https://github.com/docker-mailserver/docker-mailserver/blob/377bdb390e87868617d45fe2c086a93a168be0fa/target/scripts/helpers/accounts.sh#L3-L7

https://github.com/docker-mailserver/docker-mailserver/blob/377bdb390e87868617d45fe2c086a93a168be0fa/target/scripts/helpers/accounts.sh#L39-L41

https://github.com/docker-mailserver/docker-mailserver/blob/377bdb390e87868617d45fe2c086a93a168be0fa/target/scripts/helpers/accounts.sh#L90

And:

https://github.com/docker-mailserver/docker-mailserver/blob/65622c56e934459beef7fba045f23ef897a6dcda/target/postfix/main.cf#L12

Presumably if you set `hostname: mail.example.com` instead of `hostname: mail` + `domainname: example.com` this is the problem?

---

AFAIK `mail` is only meant to be an MX record for `@example.com` addresses, but users are either mistakenly understanding that as being tied to the domain-part of an e-mail address such as `@mail.example.com`, or they actually want that for their domain-part (in which case `mx.example.com` as the MX record would work just as well?).

IIRC [we're using `$HOSTNAME` / `hostname -f` to get FQDN info for internal reference](https://github.com/docker-mailserver/docker-mailserver/blob/377bdb390e87868617d45fe2c086a93a168be0fa/target/scripts/helpers/dns.sh#L23) instead of our own ENV like `DMS_FQDN`, but software like Postfix is also reading the OS ENV `$HOSTNAME` to set it's `myhostname` variable and in some contexts that conflicts? Technically the hostname should be a single DNS label (`mail`), and we don't want to use `domainname` either (_as it's an NIS domain, not related to the users desired FQDN_).

**Comment by geilmaker** [comment]
I was only getting this warnings, when receiving or sending (I dont really know which one of them) emails to/from the mailbox, on startup I didnt got that warning too.

```BATCH
warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains
```

And I use ```hostname: mail``` + ```domainname: domain.de``` at the moment, without an override inside the env.

**Comment by polarathene** [comment]
> I had the mailserver running on mail.domain.de and so I created my mailboxes like "[me@mail.domain.de](mailto:me@mail.domain.de)", but at that moment this domain (mail.domain.de) gets added as vhost and from that on I get this warning, like mentioned in the warning text.
> I added "[me@domain.de](mailto:me@domain.de)", so this domain got added as vhost and it works just fine!
> There seems to be a problem when adding mailboxes with the same domain/host the mailserver is running on

@geilmaker is `hostname: mail.domain.de` vs `hostname: mail` + `domainname: domain.de` no different for you? (_**EDIT:** Your latest comment confirms now there is no difference_)

Perhaps both are generating the same `HOSTNAME` environment variable (Docker handles it from the config), and it's only regarding the combination of that with the email account creation, but then I'm confused as @casperklein shows that does not reproduce the same problem.

**Comment by geilmaker** [comment]
Like mentioned above, on startup I had the same output like he got.

My output never changed on startup, it was always like:
> ```batchfile
> Attaching to mailserver
> mailserver    | [ TASKLOG ]  Welcome to docker-mailserver 10.5.0
> mailserver    | [ TASKLOG ]  Initializing setup
> mailserver    | [ TASKLOG ]  Checking configuration
> mailserver    | [ TASKLOG ]  Configuring mail server
> mailserver    | [ TASKLOG ]  Post-configuration checks
> mailserver    | [ TASKLOG ]  Starting daemons & mail server
> mailserver    | cron: started
> mailserver    | rsyslog: started
> mailserver    | dovecot: started
> mailserver    | update-check: started
> mailserver    | opendkim: started
> mailserver    | opendmarc: started
> mailserver    | postgrey: started
> mailserver    | postfix: started
> mailserver    | clamav: started
> mailserver    | changedetector: started
> mailserver    | amavis: started
> mailserver    | [ TASKLOG ]  mail.domain.de is up and running
> mailserver    | Apr 19 22:09:41 mail amavis[524]: starting. /usr/sbin/amavisd-new at mail.domain.de amavisd-new-2.11.1 (20181009), Unicode aware, LC_CTYPE="C.UTF-8"
> mailserver    | Apr 19 22:09:41 mail amavis[524]: perl=5.032001, user=, EUID: 112 (112);  group=, EGID: 114 114 (114 114)
> mailserver    | Apr 19 22:09:41 mail amavis[524]: Net::Server: Group Not Defined.  Defaulting to EGID '114 114'
> mailserver    | Apr 19 22:09:41 mail amavis[524]: Net::Server: User Not Defined.  Defaulting to EUID '112'
> mailserver    | Apr 19 22:09:41 mail amavis[524]: No ext program for   .zoo, tried: zoo
> mailserver    | Apr 19 22:09:41 mail amavis[524]: No ext program for   .doc, tried: ripole
> mailserver    | Apr 19 22:09:41 mail amavis[524]: No decoder for       .F
> mailserver    | Apr 19 22:09:41 mail amavis[524]: No decoder for       .doc
> mailserver    | Apr 19 22:09:41 mail amavis[524]: No decoder for       .zoo
> mailserver    | Apr 19 22:09:41 mail amavis[524]: Using primary internal av scanner code for ClamAV-clamd
> mailserver    | Apr 19 22:09:41 mail amavis[524]: Found secondary av scanner ClamAV-clamscan at /usr/bin/clamscan
> mailserver    | Apr 19 22:09:42 mail postfix/master[1428]: daemon started -- version 3.5.6, configuration /etc/postfix
> ```

I was just getting the warning message when sending/receiving emails, I dont know if @casperklein also tried that out. I just created 2 mailboxes (me@mail.domain.de and test@mail.domain.de), connected via IMAP with Thunderbird and then tried to send an email from me to test.

**Comment by casperklein** [comment]
I didn't send any emails, as I thought the warning came on DMS startup.

**Comment by geilmaker** [comment]
Hey guys,
I did some more tests on that one....

Now I got 2 mailboxes added:
```BATCH
me@domain.de
test@mail.domain.de
```
My Mailserver is still running on host ```mail.domain.de```.

The mailbox ```me@domain.de``` is working fine, I can send and receive emails without any problems.
The mailbox ```test@mail.domain.de``` can send emails but cannot receive any. As soon there is an email coming for that mailbox im getting the good old warning (```warning: do not list domain mail.domain.de in BOTH mydestination and virtual_mailbox_domains```) and the email can not be delivered.
